### PR TITLE
refactor(llc): deprecate `MessageSendingStatus` in favor of `MessageState`.

### PR DESCRIPTION
**Pull request by xsahil03x**
Introduces new `MessageState`, fixes retry-queue.

### CODE REVIEW

File: packages/stream_chat/lib/src/client/channel.dart [hunk 1]
```diff
@@ -582,7 +582,7 @@ class Channel {
     // Eg. Updating the message while the previous call is in progress.
     _messageAttachmentsUploadCompleter
         .remove(message.id)
-        ?.completeError('Message Cancelled');
+        ?.completeError(const StreamChatError('Message deleted'));
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/client/channel.dart:585]:
```suggestion
        ?.completeError(const StreamChatError('Message cancelled'));
```

File: packages/stream_chat/lib/src/client/channel.dart [hunk 2]
```diff
@@ -653,17 +662,18 @@ class Channel {
...
   }) async {
+    _checkInitialized();
     final originalMessage = message;
 
     // Cancelling previous completer in case it's called again in the process
     // Eg. Updating the message while the previous call is in progress.
     _messageAttachmentsUploadCompleter
         .remove(message.id)
-        ?.completeError('Message Cancelled');
+        ?.completeError(const StreamChatError('Message deleted'));
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/client/channel.dart:672]:
```suggestion
        ?.completeError(const StreamChatError('Message cancelled'));
```

File: packages/stream_chat/lib/src/client/channel.dart [hunk 3]
```diff
@@ -729,6 +750,23 @@ class Channel {
...
     bool skipEnrichUrl = false,
   }) async {
+    _checkInitialized();
+    final originalMessage = message;
+
+    // Cancelling previous completer in case it's called again in the process
+    // Eg. Updating the message while the previous call is in progress.
+    _messageAttachmentsUploadCompleter
+        .remove(message.id)
+        ?.completeError(const StreamChatError('Message deleted'));
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/client/channel.dart:760]:
```suggestion
        ?.completeError(const StreamChatError('Message cancelled'));
```

File: packages/stream_chat_flutter/lib/src/attachment/attachment_upload_state_builder.dart [hunk 4]
```diff
@@ -216,7 +216,7 @@ class _FailedState extends StatelessWidget {
                 horizontal: 12,
               ),
               child: Text(
-                context.translations.uploadErrorLabel,
+                error ?? 'Failed to upload file',
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/attachment/attachment_upload_state_builder.dart:219]:
```suggestion
                context.translations.uploadErrorLabel,
```

File: packages/stream_chat/lib/src/core/models/message_state.freezed.dart
Comment (kanat) [inline packages/stream_chat/lib/src/core/models/message_state.freezed.dart:1]: Just curios how did we generate this one? What is the source?

File: packages/stream_chat/lib/src/core/models/message_state.g.dart
Comment (kanat) [inline packages/stream_chat/lib/src/core/models/message_state.g.dart:1]: Same here =]

File: packages/stream_chat/lib/src/core/models/message_state.freezed.dart
Comment (kanat) [reply packages/stream_chat/lib/src/core/models/message_state.freezed.dart:1]: The state hierarchy is hard to read/navigate 🥲

**Review by kanat — APPROVED** [review]
Overall, LGTM

**Pull request merged**